tests/integration/targets/dynamodb_table/tasks/main.yml

```diff
@@ -0,0 +1,207 @@
+    - name: test failure with no parameters
+      dynamodb_table:
+      register: result
+      ignore_errors: yes
+
+    - name: assert failure with no parameters
+      assert:
+        that:
+          - result is failed
+          - 'result.msg == "missing required arguments: name"'
+
+    - name: test credentials from environment
+      dynamodb_table:
+        name: "{{ table_name }}"
+        hash_key_name: myhashkey
+      environment:
+        EC2_REGION: "{{ ec2_region }}"
+        EC2_ACCESS_KEY: bogus_access_key
+        EC2_SECRET_KEY: bogus_secret_key
+      register: result
+      ignore_errors: yes
+
+    - name: assert dynamodb_table with valid ec2_url
+      assert:
+        that:
+          - result is failed
+          - >
+            "Authentication failure" in result.msg
+            or "token included in the request is invalid" in result.msg
+
+    # ============================================================
+    - name: test credential parameters
+      dynamodb_table:
+        name: "{{ table_name }}"
+        hash_key_name: myhashkey
+        ec2_region: "{{ ec2_region }}"
+        ec2_access_key: bogus_access_key
+        ec2_secret_key: bogus_secret_key
+        security_token: "{{ security_token | d(omit) }}"
+      register: result
+      ignore_errors: yes
+    - name: assert credential parameters
+      assert:
+        that:
+          - result is failed
+          - >
+            "Authentication failure" in result.msg
+            or "token included in the request is invalid" in result.msg
+
+- name: dynamodb table integration tests
+  module_defaults:
+    dynamodb_table:
+      aws_access_key: "{{ aws_access_key }}"
+      aws_secret_key: "{{ aws_secret_key }}"
+      region: "{{ aws_region }}"
+      security_token: "{{ security_token | d(omit) }}"
+  block:
+    # ============================================================
+    - name: create table
+      dynamodb_table:
+        name: "{{ table_name }}"
+        hash_key_name: myhashkey
+        range_key_name: myrangekey
+        state: present
+        indexes:
+          - name: NamedIndex
+            type: global_all
+            hash_key_name: myhashkey
+          - name: NamedIndexV2
+            type: global_all
+            hash_key_name: myhashkey
+            range_key_name: myrangekey
+          - name: NamedLocalIndex
+            type: keys_only
+            hash_key_name: myhashkey
+            range_key_name: mylocalrangekey
+        tags:
+          First: one
+          Second: two
+      register: result
+      until:
+        - result | json_query('global_secondary_indexes[?IndexName==`NamedIndexV2`].IndexStatus') == ['ACTIVE']
+        - result | json_query('global_secondary_indexes[?IndexName==`NamedIndex`].IndexStatus') == ['ACTIVE']
+      retries: 6
+      delay: 10
+
+    - name: update index on the existing table
+      dynamodb_table:
+        name: "{{ table_name }}"
+        hash_key_name: myhashkey
+        indexes:
```

Comment:
AWS permits only one index to be created in a single action, this task is throwing an unhandled ClientError.  The code should reflect and enforce the service limits.
`botocore.errorfactory.LimitExceededException: An error occurred (LimitExceededException) when calling the UpdateTable operation: Subscriber limit exceeded: Only 1 online index can be created or deleted simultaneously per table`
https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/Limits.html#limits-secondary-indexes

I haven't done a proper review of the code, but there doesn't seem to be much exception handling.  We generally like to catch `ClientError` and `BotoCoreError`.